fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -2041,41 +2047,135 @@ ACTOR Future<Void> TagPartitionedLogSystem::getDurableVersionChanged(LogLockInfo
 	return Void();
 }
 
+void getTLogLocIds(std::vector<Reference<LogSet>>& tLogs,
+                   std::vector<std::tuple<int, std::vector<TLogLockResult>>>& logGroupResults,
+                   std::vector<std::vector<uint16_t>>& tLogLocIds,
+                   uint16_t& maxTLogLocId) {
+	// Initialization.
+	tLogLocIds.clear();
+	tLogLocIds.resize(logGroupResults.size());
+	maxTLogLocId = 0;
+
+	// Map the interfaces of all (local) tLogs to their corresponding locations in LogSets.
+	std::map<UID, uint16_t> interfLocMap;
+	uint16_t location = 0;
+	for (auto& it : tLogs) {
+		if (!it->isLocal) {
+			continue;
+		}
+		for (uint16_t i = 0; i < it->logServers.size(); i++) {
+			interfLocMap[it->logServers[i]->get().interf().id()] = location++;
```

Comment:
should add `if (it->logServers[i]->get().present()) {` here, per earlier discussion.

Reply:
Agreed. This is a cherry-pick, so let's not make that change in this PR?